scp.c

```diff
@@ -1309,9 +1327,20 @@ syserr:			run_err("%s: %s", name, strerror(errno));
 				goto next;
 		}
 #define	FILEMODEMASK	(S_ISUID|S_ISGID|S_IRWXU|S_IRWXG|S_IRWXO)
+#ifdef WINDOWS
+		buf_len = strlen(last) * 2;
```

Comment:
can you verify this change by having non-english characters in the foldernames and filenames...